networkx/algorithms/centrality/betweenness.py

```diff
@@ -236,6 +236,11 @@ def edge_betweenness_centrality(G, k=None, normalized=True, weight=None, seed=No
     betweenness = _rescale_e(
         betweenness, len(G), normalized=normalized, directed=G.is_directed()
     )
+    if G.is_multigraph():
+        betweenness = _add_edge_keys(G, betweenness, weight=weight)
+        for e in G.edges():  # Remove edges without key
```

Comment:
Can you help me understand why we need to remove the edges in `betweenness` that don't have a key?
Do some edges have keys and other don't?  Are we sure we don't throw out any edges that we might actually want?  e.g. what if there is only one edge between a pair of nodes. Will this throw out the betweenness for that edge?

Reply:
The reasoning behind removing those edges lies in my aim to comply with the first example given for [```set_edge_attributes```](https://networkx.org/documentation/stable/reference/generated/networkx.classes.function.set_edge_attributes.html). I think it shouldn't be possible (especially for novices) to encounter an example that throws an error. Leaving them in would cause ```
ValueError: not enough values to unpack (expected 3, got 2)``` in ```networkx.functions``` [line 811](https://github.com/pinselimo/networkx/blob/add-multigraph-betweenness/networkx/classes/function.py#L811).

There are no edges without a key. Adding an edge on a ```MultiGraph``` using ```add_edge``` with ```key=None``` (the default value) will create and add a new key. This is also the return value of ```add_edge``` in ```MultiGraph```s. (Same goes for ```MultiDiGraph```. ```add_edges_from``` uses ```add_edge``` internally).

I initially added it outside the function because in [line 234](https://github.com/pinselimo/networkx/blob/add-multigraph-betweenness/networkx/algorithms/centrality/betweenness.py#L234) nodes are removed in the same scope. But I will move the code inside the function call.